addons/l10n_it_edi/models/account_move_send.py

```diff
@@ -100,12 +104,19 @@ def _link_invoice_documents(self, invoices_data):
         # EXTENDS 'account'
         super()._link_invoice_documents(invoices_data)
 
-        attachments_vals = [
-            invoice_data.get('l10n_it_edi_values')
-            for invoice_data in invoices_data.values()
-            if invoice_data.get('l10n_it_edi_values')
-        ]
-        if attachments_vals:
-            attachments = self.env['ir.attachment'].sudo().create(attachments_vals)
-            res_ids = attachments.mapped('res_id')
-            self.env['account.move'].browse(res_ids).invalidate_recordset(fnames=['l10n_it_edi_attachment_id', 'l10n_it_edi_attachment_file'])
+        moves_to_names = {}
+        for move, data in invoices_data.items():
+            if data.get('l10n_it_edi_values'):
+                move.l10n_it_edi_attachment_file = base64.b64encode(data['l10n_it_edi_values']['raw'])
+                move.l10n_it_edi_attachment_name = data['l10n_it_edi_values']['name']
+                move.invalidate_recordset(fnames=['l10n_it_edi_attachment_name', 'l10n_it_edi_attachment_file'])
+                moves_to_names[move.id] = data['l10n_it_edi_values']['name']
+
+        if moves_to_names:
+            attachments = self.env['ir.attachment'].sudo().search([
+                ('res_model', '=', 'account.move'),
+                ('res_field', '=', 'l10n_it_edi_attachment_file'),
+                ('res_id', 'in', list(moves_to_names.keys())),
+            ])
+            for attachment in attachments:
+                attachment.name = moves_to_names.get(attachment.res_id)
```

Comment:
```suggestion
        attachments = self.env['ir.attachment'].search([
            ('res_model', '=', 'account.move'),
            ('res_field', '=', 'l10n_it_edi_attachment_file'),
            ('res_id', 'in', [invoice.id for invoice in invoices_data]),
        ]).grouped('res_id')
        
        for move, data in invoices_data.items():
            if values := data.get('l10n_it_edi_values'):
                move.l10n_it_edi_attachment_file = base64.b64encode(values['raw'])
                move.l10n_it_edi_attachment_name = values['name']
                move.invalidate_recordset(fnames=['l10n_it_edi_attachment_name', 'l10n_it_edi_attachment_file'])
                if attachment := attachments.get(move.id):
                    attachment.name = values['name']
```
A bit shorter, let me know what you think (not tested)?

Reply:
I reverted back to the original since I couldn't find a way to change the attachment name when a value is assigned to the binary field. TIL that `self.name` is used both for the attachment name and res_field when the attachment is created from a binary field, so I think using the dictionary here would be better :)
![image](https://github.com/user-attachments/assets/7838c9b4-23c1-4192-850f-67f11b460d00)